flax/nnx/pytreelib.py

```diff
@@ -936,7 +936,11 @@ def _graph_node_set_key(self, key: str, value: tp.Any):
   def _graph_node_pop_key(self, key: str):
     if not isinstance(key, str):
       raise KeyError(f'Invalid key: {key!r}')
-    return vars(self).pop(key)
+    value = vars(self).pop(key)
```

Comment:
maybe we should use `delatter` and override `__delattr__` to make sure attribute deletion in general is kept in sync with `_pytree__nodes`